x-pack/plugins/security/public/management/api_keys/api_keys_grid/api_keys_table.tsx

```diff
@@ -0,0 +1,635 @@
+import {
+  EuiBadge,
+  EuiBasicTable,
+  EuiButton,
+  EuiFilterButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHealth,
+  EuiLink,
+  EuiSearchBar,
+  EuiSpacer,
+  EuiText,
+  EuiToolTip,
+} from '@elastic/eui';
+import moment from 'moment-timezone';
+import type { FunctionComponent } from 'react';
+import React, { useState } from 'react';
+
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { UserAvatar, UserProfilesPopover } from '@kbn/user-profile-components';
+
+import type { AuthenticatedUser } from '../../../../common';
+import type { ApiKey, ApiKeyAggregations, RestApiKey } from '../../../../common/model';
+import type { CreateAPIKeyResult } from '../api_keys_api_client';
+
+export interface ApiKeysTableProps {
+  apiKeys: CategorizedApiKey[];
+  currentUser: AuthenticatedUser;
+  createdApiKey?: CreateAPIKeyResult;
+  readOnly?: boolean;
+  loading?: boolean;
+  canManageCrossClusterApiKeys: boolean;
+  canManageApiKeys: boolean;
+  canManageOwnApiKeys: boolean;
+  onClick(apiKey: CategorizedApiKey): void;
+  onDelete(apiKeys: CategorizedApiKey[]): void;
+  totalItemCount?: number;
+  onTableChange: any;
+  pagination: any;
+  onSearchChange: any;
+  aggregations?: ApiKeyAggregations;
+}
+
+export const ApiKeysTable: FunctionComponent<ApiKeysTableProps> = ({
+  apiKeys,
+  createdApiKey,
+  currentUser,
+  onClick,
+  onDelete,
+  canManageApiKeys = false,
+  canManageOwnApiKeys = false,
+  readOnly = false,
+  loading = false,
+  totalItemCount = 0,
+  onTableChange,
+  pagination,
+  onSearchChange,
+  aggregations,
+}) => {
+  const columns: Array<EuiBasicTableColumn<CategorizedApiKey>> = [];
+  const [selectedItems, setSelectedItems] = useState<CategorizedApiKey[]>([]);
+  const initialQuery = EuiSearchBar.Query.parse('');
+
+  const { typeFilters, usernameFilters, expired } = categorizeAggregations(aggregations);
+
+  const deletable = (item: CategorizedApiKey) =>
+    canManageApiKeys || (canManageOwnApiKeys && item.username === currentUser.username);
+
+  columns.push(
+    {
+      field: 'name',
+      name: i18n.translate('xpack.security.management.apiKeys.table.nameColumnName', {
+        defaultMessage: 'Name',
+      }),
+      sortable: true,
+      render: (name: string, item: CategorizedApiKey) => {
+        return (
+          <EuiLink onClick={() => onClick(item)} data-test-subj={`apiKeyRowName-${item.name}`}>
+            {name}
+          </EuiLink>
+        );
+      },
+    },
+    {
+      field: 'type',
+      name: (
+        <FormattedMessage
+          id="xpack.security.management.apiKeys.table.typeColumnName"
+          defaultMessage="Type"
+        />
+      ),
+      sortable: true,
+      render: (type: CategorizedApiKey['type'], apiKeyRecord) => {
+        let keyType = type;
+        if (
+          apiKeyRecord.name.indexOf('Alerting: ') === 0 ||
+          apiKeyRecord.metadata?.managed === true
+        ) {
+          keyType = 'managed';
+        }
```

Comment:
__issue:__ it seems we're using different conditions in UI and on the server-side for alerting keys. I missed the space in my key name and now server returns it as managed, but UI labels it as personal:
![Screenshot from 2024-03-27 11-09-28](https://github.com/elastic/kibana/assets/1713708/c3d80e38-1724-4f2a-a2fa-ce38266167c6)